stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -372,7 +373,8 @@ function _rdiv!(B::AbstractVecOrMat, A::AbstractVecOrMat, D::Diagonal)
     B
 end
 
-\(D::Diagonal, B::AbstractVecOrMat) = ldiv!(similar(B, promote_op(\, eltype(D), eltype(B)), size(B)), D, B)
+\(D::Diagonal, B::AbstractVecOrMat) =
```

Comment:
This change broke StaticArrays tests:
```julia
using StaticArrays, LinearAlgebra

m2 = diagm(0 => [11, 12, 13, 14])
b = @SVector [2,-1,2,1]
m2\b # throws now
```
See also here: https://github.com/JuliaArrays/StaticArrays.jl/runs/5032815685?check_suite_focus=true

Reply:
Thanks for reporting and sorry about the inconvenience. I think I know how to fix this. This method is challenging: we can't use 3-arg `similar` because destroys special structure, and we can't use 2-arg `similar` because that returns `Hermitian` for `Hermitian` input, which we can't write into. Also, `Diagonal*Hermition` is not `Hermitian`, so this is wrong in any case. So I think we need to expand this and avoid the call to `ldiv!`.

Reply:
No problem, I sometimes check nightly test failures to catch such issues. Unfortunately I don't know what the right way to solve this is.

Reply:
The should be in #44018, I'm running nanosoldier to check.